src/common/RadioButton.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <app-button
+    color="none"
+    size="none"
+    :class="radioClasses"
+    :text="option.label"
+    @click="emitSelection"
+  >
+    <template #nodeLeft>
+      <div class="radio-button__circle" />
+    </template>
+  </app-button>
+</template>
+
+<script lang="ts" setup>
+import { computed } from 'vue'
+import { AppButton } from '@/common'
+
+const props = defineProps<{
+  modelValue: string
+  option: { label: string; value: string }
+  isChosen: boolean
+}>()
+
+const emit = defineEmits<{
+  (e: 'update:modelValue', value: string): void
+}>()
+
+const radioClasses = computed(() =>
+  [
+    'radio-button',
+    ...(props.isChosen ? ['radio-button--active'] : []),
+    ...(props.modelValue ? ['radio-button--chosen'] : []),
```

Comment:
Also it is considered as a better practice to rely on `aria` attributes in such cases instead of class modifiers. It provides better semantics.

**hmtl**:
```html
<app-button
    color="none"
    size="none"
    :aria-selected="Boolean(modelValue)"
    :class="radioClasses"
    :text="option.label"
    @click="emitSelection"
  >
```

**css**:
```scss
.radio-button {
    &[aria-selected='true'] {
        /* whatever style you need */
    }
}
```

Reply:
That's an excellent point, fixed